### PR TITLE
Chore/upgrade golang and remove pkg errors

### PR DESCRIPTION
**Pull request by ravilock**

### CODE REVIEW

**Review by wpjunior — APPROVED** [review]

File: internal/provider/resource_tsuru_app_deploy_test.go [hunk 1]
```diff
@@ -58,6 +62,7 @@ func TestAccResourceTsuruAppDeploy(t *testing.T) {
 	}
 	server := httptest.NewServer(fakeServer)
 	os.Setenv("TSURU_TARGET", server.URL)
+	os.Setenv("TSURU_TOKEN", "testing-token")
```
Comment (crgisch) [inline internal/provider/resource_tsuru_app_deploy_test.go]: Can set the env for the Job Deploy test aswell (which is most likely working now because of the Setenv leaking to the next tests)

File: internal/provider/resource_tsuru_app_deploy.go [hunk 2]
```diff
@@ -114,14 +113,13 @@ func resourceTsuruApplicationDeployDo(ctx context.Context, d *schema.ResourceDat
 
 	token := provider.Token
 	if token == "" {
-		token = deployToken()
+		return diag.Errorf("token not available")
```
Comment (crgisch) [inline internal/provider/resource_tsuru_app_deploy.go:116]: can add a test for when token is empty

**Review by crgisch — CHANGES_REQUESTED** [review]

**Review by crgisch — APPROVED** [review]
LGTM 👍
